vmdb/app/models/ems_kubernetes.rb

```diff
@@ -13,17 +13,41 @@ def self.description
 
   def self.raw_connect(hostname, port, api_version)
     require 'kubeclient'
-    require 'uri'
-
-    uri = URI::HTTP.build(:path => "/api", :port => port.to_i)
+    api_endpoint = raw_api_endpoint(hostname, port)
+    Kubeclient::Client.new(api_endpoint, api_version)
+  end
 
+  def self.raw_api_endpoint(hostname, port)
+    require 'uri'
     # URI::Generic#hostname= was added in ruby 1.9.3 and will automatically
     # wrap an ipv6 address in []
-    uri.hostname = hostname
-    Kubeclient::Client.new uri.to_s, api_version
+    URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/api')
```

Comment:
No this code changed it from calling .hostname to passing via Hash.  I
think the dot notation is required for ipv6 support.  @jrafanie ?
 On Feb 19, 2015 1:12 PM, "Federico Simoncelli" <notifications@github.com>
wrote:

> In vmdb/app/models/ems_kubernetes.rb
> <https://github.com/ManageIQ/manageiq/pull/1679#discussion_r25010291>:
>
> >      # URI::Generic#hostname= was added in ruby 1.9.3 and will automatically
> >      # wrap an ipv6 address in []
> > -    uri.hostname = hostname
> > -    Kubeclient::Client.new uri.to_s, api_version
> > +    URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/api')
>
> Is it possible that it's unrelated to my refactoring? I don't think I
> changed anything in that regard and if it's a separate improvement maybe
> doesn't belong to my patch.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/manageiq/pull/1679/files#r25010291>.
>